Python/Product/Analysis/LanguageServer/DiagnosticsErrorSink.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PythonTools.Analysis.Infrastructure;
+using Microsoft.PythonTools.Parsing;
+
+namespace Microsoft.PythonTools.Analysis.LanguageServer {
+    class DiagnosticsErrorSink : ErrorSink {
+        private readonly string _source;
+        private readonly List<Diagnostic> _diagnostics;
+        private readonly IReadOnlyList<KeyValuePair<string, DiagnosticSeverity>> _taskCommentMap;
+
+        public DiagnosticsErrorSink(string source, List<Diagnostic> diagnostics, IReadOnlyDictionary<string, DiagnosticSeverity> taskCommentMap = null) {
```

Comment:
`DiagnosticsErrorSink` is called in two places and both times `diagnostics = new List<Diagnostic>()`, so you can drop this parameter.

Reply:
That actually changes the semantics away from what I want this class to be. I'll change the argument to `Action<Diagnostic>` and remove the (unused) Diagnostics property.